packages/frontend-2/components/settings/workspaces/General.vue

```diff
@@ -0,0 +1,143 @@
+            Delete workspace
+          </FormButton>
+        </div>
+      </div>
+    </div>
+    <SettingsWorkspacesGeneralDeleteDialog
+      v-if="workspaceResult"
+      v-model:open="showDeleteDialog"
+      :workspace="workspaceResult.workspace"
+    />
+  </section>
+</template>
+
+<script setup lang="ts">
+import { graphql } from '~~/lib/common/generated/gql'
+import { useForm } from 'vee-validate'
+import { useActiveUser } from '~/lib/auth/composables/activeUser'
+import { Roles } from '@speckle/shared'
+import { useQuery, useMutation } from '@vue/apollo-composable'
+import { settingsUpdateWorkspaceMutation } from '~/lib/settings/graphql/mutations'
+import { settingsWorkspaceGeneralQuery } from '~/lib/settings/graphql/queries'
+import type { WorkspaceUpdateInput } from '~~/lib/common/generated/gql/graphql'
+import { ToastNotificationType, useGlobalToast } from '~~/lib/common/composables/toast'
+import {
+  getFirstErrorMessage,
+  convertThrowIntoFetchResult
+} from '~~/lib/common/helpers/graphql'
+import { isRequired, isStringOfLength } from '~~/lib/common/helpers/validation'
+
+graphql(`
+  fragment SettingsWorkspacesGeneral_Workspace on Workspace {
+    ...SettingsWorkspaceGeneralDeleteDialog_Workspace
+    logo
+    description
+  }
+`)
+
+type FormValues = { name: string; description: string }
+
+const props = defineProps<{
+  workspaceId: string
+}>()
+
+const { activeUser: user } = useActiveUser()
+const { handleSubmit } = useForm<FormValues>()
+const { triggerNotification } = useGlobalToast()
+const { mutate: updateMutation } = useMutation(settingsUpdateWorkspaceMutation)
+const { result: workspaceResult } = useQuery(settingsWorkspaceGeneralQuery, () => ({
+  id: props.workspaceId
+}))
+
+const name = ref('')
+const description = ref('')
+const showDeleteDialog = ref(false)
+
+const isAdmin = computed(() => user.value?.role === Roles.Server.Admin)
+
+const save = handleSubmit(async () => {
+  if (!workspaceResult.value?.workspace) return
+
+  const input: WorkspaceUpdateInput = {
+    id: workspaceResult.value.workspace.id
+  }
+  if (name.value !== workspaceResult.value.workspace.name) input.name = name.value
+  if (description.value !== workspaceResult.value.workspace.description)
+    input.description = description.value
+
+  const result = await updateMutation({ input }).catch(convertThrowIntoFetchResult)
+
+  if (result?.data) {
+    triggerNotification({
+      type: ToastNotificationType.Success,
+      title: 'Workspace updated'
+    })
+  } else {
+    const errorMessage = getFirstErrorMessage(result?.errors)
+    triggerNotification({
+      type: ToastNotificationType.Danger,
+      title: 'Workspace update failed',
+      description: errorMessage
+    })
+  }
+})
+
+function openDeleteDialog() {
+  showDeleteDialog.value = true
+}
+
+watch(
+  () => workspaceResult,
+  () => {
+    if (workspaceResult.value?.workspace) {
+      name.value = workspaceResult.value.workspace.name
```

Comment:
You're also asking for the workspace ID and workspace name, and yet these don't show up in the fragment. If the DeleteDialog removes those requirements, this seemingly unrelated component will break